internal/controller/etcdcluster_controller.go

```diff
@@ -52,10 +54,12 @@ type EtcdClusterReconciler struct {
 // Every phase of Reconcile stores intermediate information here so that
 // subsequent phases can operate without additional lookups.
 type reconcileState struct {
-	cluster        *ecv1alpha1.EtcdCluster      // cluster custom resource currently being reconciled
-	sts            *appsv1.StatefulSet          // associated StatefulSet for the cluster
-	memberListResp *clientv3.MemberListResponse // member list fetched from the etcd cluster
-	memberHealth   []etcdutils.EpHealth         // health information for each etcd member
+	cluster          *ecv1alpha1.EtcdCluster       // cluster custom resource currently being reconciled
+	sts              *appsv1.StatefulSet           // associated StatefulSet for the cluster
+	memberListResp   *clientv3.MemberListResponse  // member list fetched from the etcd cluster
+	memberHealth     []etcdutils.EpHealth          // health information for each etcd member
+	calculatedStatus *ecv1alpha1.EtcdClusterStatus // calculated status for this reconcile cycle
```

Comment:
we discussed during the zoom call of 25nov that this shouldn't be needed.

normally the way you reconcile status, canonically in k8s is to:
- read the status you want by e.g. comparing to spec
- read the existing status
- create a patch
- apply the patch in single take

you normally wouldn't need to store the status in a variable